aea/cli/scaffold.py

```diff
@@ -360,3 +366,47 @@ def _scaffold_non_package_item(
     except Exception as e:
         os.remove(dest)
         raise click.ClickException(str(e))
+
+
+def add_contract_abi(ctx: Context, contract_name: str, contract_abi_path: str) -> None:
+    """
+    Add the contract ABI to a contract scaffold.
+
+    :param ctx: the CLI context.
+    :param contract_name: the contract name.
+    :param contract_abi_path: the contract ABI path.
+    """
+
+    # Get some data from the context
+    to_local_registry = ctx.config.get("to_local_registry")
+    author_name = ctx.agent_config.author
+
+    # Create the build directory and copy the ABI file
+    src = Path(contract_abi_path)
+
+    contract_dir_root = (
+        Path(str(ctx.agent_config.directory)) if to_local_registry else Path()
+    )
+    contract_dir = contract_dir_root / Path("contracts") / Path(contract_name)
+    abi_dest = contract_dir / Path("build") / Path(src.name)
+
+    click.echo(f"Updating contract scaffold '{contract_name}' to include ABI...")
+
+    os.makedirs(os.path.dirname(abi_dest), exist_ok=True)
+    shutil.copyfile(src, abi_dest)
+
+    # Update contract.yaml
+    with open(contract_dir / Path("contract.yaml"), "r") as contract_yaml:
+        content = yaml.full_load(contract_yaml)
+
+    content["contract_interface_paths"] = {"ethereum": f"build/{src.name}"}
+
+    with open(contract_dir / Path("contract.yaml"), "w") as contract_yaml:
+        yaml.dump(content, contract_yaml, sort_keys=False)
```

Comment:
we have specific tooling to load and dump the config, we don't do it raw like this, check aea.configuration